sdk/samples/router/create_router_with_pyfunc_ensembler.py

```diff
@@ -0,0 +1,155 @@
+from turing.router.config.log_config import LogConfig, ResultLoggerType
+from turing.router.config.router_ensembler_config import PyfuncRouterEnsemblerConfig
+from turing.router.config.experiment_config import ExperimentConfig
+
+from typing import List, Any
+
+
+# To register a pyfunc ensembler to be used in a Turing router, implement the `turing.ensembler.PyFunc` interface
+class SampleEnsembler(turing.ensembler.PyFunc):
+    """
+    A simple ensembler, that returns the value corresponding to the version that has been specified in the
+    `features` in each request. This value if obtained from the route responses found in the `predictions` in each
+    request.
+
+    If no version is specified in `features`, return the sum of all the values of all the route responses in
+    `predictions` instead.
+
+    e.g. The values in the route responses (`predictions`) corresponding to the versions, `a`, `b` and `c` are 1, 2
+         and 3 respectively.
+
+         For a given request, if the version specified in `features` is "a", the ensembler would return the value 1.
+
+         If no version is specified in `features`, the ensembler would return the value 6 (1 + 2 + 3).
+    """
+    # `initialize` is essentially a method that gets called when an object of your implemented class gets instantiated
+    def initialize(self, artifacts: dict):
+        pass
+
+    # Each time a Turing Router sends a request to a pyfunc ensembler, ensemble will be called, with the request payload
+    # being passed as the `features` argument, and the route responses as the `predictions` argument.
+    #
+    # If an experiment has been set up, the experiment returned would also be passed as the `treatment_config` argument.
+    #
+    # The return value of `ensemble` will then be returned as a `json` payload to the Turing router.
+    def ensemble(
+            self,
+            features: dict,
+            predictions: List[dict],
+            treatment_config: dict) -> Any:
+        # Get a mapping between route names and their corresponding responses
+        routes_to_response = dict()
+        for prediction in predictions:
+            routes_to_response[prediction["route"]] = prediction
+
+        if "version" in features:
+            return routes_to_response[features["version"]]["data"]["value"]
+        else:
+            return sum(response["data"]["value"] for response in routes_to_response.values())
+
+
+def main(turing_api: str, project: str):
+    # Initialize Turing client
+    turing.set_url(turing_api)
+    turing.set_project(project)
+
+    # Register an ensembler with Turing:
+    ensembler = turing.PyFuncEnsembler.create(
+        name="sample-ensembler-1",
+        ensembler_instance=SampleEnsembler(),
+        conda_env={
+            'dependencies': [
+                'python>=3.7.0',
+                # other dependencies, if required
+            ]
+        }
+    )
+    print("Ensembler created:\n", ensembler)
```

Comment:
Added a new file to demonstrate the deployment of a router with a pyfunc ensembler.